rust/src/utils/distance.rs

```diff
@@ -60,12 +61,64 @@ unsafe fn euclidean_distance_fma(from: &[f32], to: &[f32]) -> f32 {
 ///
 #[inline]
 pub fn l2_distance_arrow(from: &Float32Array, to: &Float32Array) -> f32 {
-    let mul: Float32Array = binary(from, to, |a, b| (a - b).powf(2.0)).unwrap();
-    sum(&mul).unwrap()
+    let a = from.values();
+    let b = to.values();
+    let mut d = 0.0;
+    // Better chance to auto-vectorization.
```

Comment:
Only got about 5% from this.

Reply:
that's interesting. thought those blog posts showed bigger gains